src/JSON.jl

```diff
@@ -104,6 +104,67 @@ print(io::IO, obj, indent=nothing) = json(io, obj; pretty=something(indent, 0))
 print(a, indent=nothing) = print(stdout, a, indent)
 @doc (@doc json) print
 
+"""
+    JSON.writefile(Vector{UInt8}, x; kw...)::Vector{UInt8}
+    JSON.writefile(io::IO, x; kw...)
+    JSON.writefile(filename::AbstractString, x; kw...)
+
+Serialize `x` to JSON format. This function provides the same functionality as [`JSON.json`](@ref)
+
+The first method returns the JSON output as a `Vector{UInt8}`.
+The second method writes JSON output to an `IO` object.
+The third method writes JSON output to a file specified by `filename` which will
+be created if it does not exist yet or overwritten if it does exist.
+
+This function is provided for backward compatibility when upgrading from 
+older versions of JSON.jl where the `json` function signature differed.
+For new code, prefer using [`JSON.json`](@ref) directly.
+
+All methods accept the same keyword arguments as [`JSON.json`](@ref):
+
+- `omit_null`, `omit_empty`, `allownan`, `jsonlines`, `pretty`, `inline_limit`
+- `ninf`, `inf`, `nan`, `float_style`, `float_precision`, `bufsize`, `style`
+
+See [`JSON.json`](@ref) for detailed documentation of all keyword arguments and more examples.
+
+# Examples
+```julia
+# Write to IO
+io = IOBuffer()
+JSON.writefile(io, Dict("key" => "value"))
+String(take!(io))  # "{\"key\":\"value\"}"
+
+# Write to file
+JSON.writefile("output.json", [1, 2, 3])
+
+# Get as bytes
+bytes = JSON.writefile(Vector{UInt8}, Dict("hello" => "world"))
+String(bytes)  # "{\"hello\":\"world\"}"
+```
+"""
+function writefile end
+
+function writefile(::Type{Vector{UInt8}}, x; pretty::Union{Integer,Bool}=false, kw...)
```

Comment:
I'm not following why we want this or the 2nd definitions? `writefile` that doesn't write to a file and just returns a byte vector?

Reply:
I want to avoid extra conversions between `String` and `Vector{UInt8}` for use with Zarr.jl and SmallZarrGroups.jl

I'm also trying to match the style of the `writefile` function in Parquet2.jl https://expandingman.gitlab.io/Parquet2.jl/#Writing-Data

Reply:
Here is a benchmark on the cost of doing the extra String to Vector{UInt8} conversion:

```julia-repl
julia> using Chairmarks, JSON, Random

julia> @be randstring(10000) Vector{UInt8}(JSON.json(_)) seconds=2
Benchmark: 52143 samples with 4 evaluations
 min    4.807 μs (7 allocs: 19.852 KiB)
 median 4.874 μs (7 allocs: 19.852 KiB)
 mean   5.566 μs (7 allocs: 19.856 KiB, 0.82% gc time)
 max    15.658 ms (7 allocs: 19.883 KiB, 99.92% gc time)

julia> @be randstring(10000) JSON.write_json(Vector{UInt8}, _) seconds=2
Benchmark: 50390 samples with 7 evaluations
 min    2.971 μs (4 allocs: 9.914 KiB)
 median 3.006 μs (4 allocs: 9.914 KiB)
 mean   3.394 μs (4 allocs: 9.919 KiB, 0.82% gc time)
 max    8.692 ms (4 allocs: 9.945 KiB, 99.92% gc time)
```